### PR TITLE
ci: migrate `semantic-release` to `changesets`

### LINKED ISSUE #441: Maintainace: replace `semantic-release` with `changesets`
https://github.com/prettier/prettier-eslint-cli/pull/438#issuecomment-1213598375

Using `semantic-release` is really a bad experience to me these days...

#437
#439
#440

We must be very careful to draft the commit message, and we have no way to reverse.

[`changesets`](https://github.com/changesets/changesets) is much easier to use and to change after committing or merging, we will just focus the changesets files, rather than a single commit message carefully.

### PR DESCRIPTION
**Pull request by JounQin**
**What**: close #441

**Why**: Using `semantic-release` is really a bad experience to me these days...

**How**: implemented

### CODE REVIEW

File: package.json
Comment (JounQin) [inline package.json:17]: https://github.com/privatenumber/clean-pkg-json

**Review by kylemh — COMMENTED** [review]
For the uninitiated, will you add more to the changesets readme on how we should go about releasing?

**Comment by JounQin** [comment]
> For the uninitiated, will you add more to the changesets readme on how we should go about releasing?

Sure, thanks for reminding!

**Review by kylemh — APPROVED** [review]

**Comment by kachkaev** [comment]
There is an issue with using `secrets.GITHUB_TOKEN` with Changesets: https://github.com/changesets/action/issues/187#issuecomment-1220791777

Might be worth replacing with `secrets.CUSTOM_GITHUB_TOKEN` in a follow-up PR.

**Comment by JounQin** [comment]
> There is an issue with using `secrets.GITHUB_TOKEN` and Changesets: [changesets/action#187 (comment)](https://github.com/changesets/action/issues/187#issuecomment-1220791777)
> 
> Might be worth replacing with `secrets.CUSTOM_GITHUB_TOKEN` in a follow-up PR.

Thanks for pointing it out, but I didn't meat this issue previously. Of it occurrs, we can try to align.

**Comment by JounQin** [comment]
> There is an issue with using `secrets.GITHUB_TOKEN` and Changesets: [changesets/action#187 (comment)](https://github.com/changesets/action/issues/187#issuecomment-1220791777)
> 
> Might be worth replacing with `secrets.CUSTOM_GITHUB_TOKEN` in a follow-up PR.

Thanks for pointing it out, but I didn't meat this issue previously. Of it occurrs, we can try to align.

**Comment by JounQin** [comment]
> There is an issue with using `secrets.GITHUB_TOKEN` and Changesets: [changesets/action#187 (comment)](https://github.com/changesets/action/issues/187#issuecomment-1220791777)
> 
> Might be worth replacing with `secrets.CUSTOM_GITHUB_TOKEN` in a follow-up PR.

@kachkaev 

Now, I think the GitHub's default behavior is a very good feature, a PR generated by changesets do not need to run CI tasks IMO, they just remove changesets files, change `CHANGELOG.md` and update `package.json`, there seems nothing to check on CI, HDYT?

**Comment by kachkaev** [comment]
I see what you mean. Not running GitHub Actions for ‘Version packages’ PR will save some compute time indeed. There is pretty much zero chance for those checks to fail, so not running them may make sense.

The situation changes if at least one PR check is required:

Screenshot 2022-08-25 at 12 09 20

This will block the merge button.

This repo does not seem to have any required PR checks, so using `CUSTOM_GITHUB_TOKEN` does not seem necessary, at least for now.

**Comment by JounQin** [comment]
> The situation changes if at least one PR check is required

I met this situation previously indeed, and it's a @github's production issue IMO.

**Pull request merged**
